test/e2e-sno-disruptive/sno_disruptive_test.go

```diff
@@ -1,7 +1,261 @@
+			return false, nil /*retry*/
+		}
+		if !v1helpers.IsOperatorConditionFalse(ckaso.Status.Conditions, "StaticPodFallbackRevisionDegraded") {
+			t.Logf("StaticPodFallbackRevisionDegraded condition is set")
+			return false, nil /*retry*/
+		}
+
+		if time.Since(startTs) > mustBeReadyFor {
+			t.Logf("The cluster has been in good condition for %s", mustBeReadyFor.String())
+			return true, nil /*done*/
+		}
+		return false, nil /*wait a bit more*/
+	})
+}
+
+func setUnsupportedConfig(t testing.TB, cs clientSet, cfg map[string][]string) {
+	t.Helper()
+
+	t.Logf("Setting UnsupportedConfigOverrides to %v", cfg)
+	err := retry.OnError(retry.DefaultRetry, func(error) bool { return true }, func() error {
+		raw, err := json.Marshal(cfg)
+		if err != nil {
+			t.Log(err)
+			return err
+		}
+		ckaso, err := cs.Operator.Get(context.TODO(), "cluster", metav1.GetOptions{})
+		if err != nil {
+			t.Log(err)
+			return err
+		}
+		ckaso.Spec.UnsupportedConfigOverrides.Raw = raw
+		_, err = cs.Operator.Update(context.TODO(), ckaso, metav1.UpdateOptions{})
+		if err != nil {
+			t.Log(err)
+		}
+		return err
+	})
+	require.NoError(t, err)
+}
+
+func waitForFallbackDegradedCondition(t testing.TB, cs clientSet) {
+	t.Helper()
+
+	t.Log("Waiting for StaticPodFallbackRevisionDegraded condition, interval = 20s, timeout = 6min")
+	err := wait.Poll(20*time.Second, 6*time.Minute, func() (bool, error) {
+		ckaso, err := cs.Operator.Get(context.TODO(), "cluster", metav1.GetOptions{})
+		if err != nil {
+			t.Logf("unable to get kube-apiserver-operator resource: %v", err)
+			return false, nil /*retry*/
+		}
+
+		if v1helpers.IsOperatorConditionFalse(ckaso.Status.Conditions, "StaticPodFallbackRevisionDegraded") {
+			t.Logf("StaticPodFallbackRevisionDegraded condition hasn't been set yet")
+			return false, nil /*retry*/
+		}
+
+		return true, nil /*done*/
+	})
+	require.NoError(t, err)
+}
+
+func assertNodeStatus(t testing.TB, cs clientSet) (string, int32) {
+	t.Helper()
+
+	t.Log("Checking if a NodeStatus has been updated to report the fallback condition")
+
+	ckaso, err := cs.Operator.Get(context.TODO(), "cluster", metav1.GetOptions{})
+	require.NoError(t, err)
+
+	var nodeName string
+	var failedRevision int32
+	for _, ns := range ckaso.Status.NodeStatuses {
+		if ns.LastFallbackCount != 0 && len(nodeName) > 0 {
+			t.Fatalf("multiple node statuses report the fallback, previously on node %v, revision %v, currently on node %v, revision %v", nodeName, failedRevision, ns.NodeName, ns.LastFailedRevision)
+		}
+		if ns.LastFallbackCount != 0 {
+			nodeName = ns.NodeName
+			failedRevision = ns.LastFailedRevision
+		}
+	}
+
+	t.Logf("The fallback has been reported on node %v, failed revision is %v", nodeName, failedRevision)
+	return nodeName, failedRevision
+}
+
+func assertKasPodAnnotatedOnNode(t testing.TB, cs clientSet, expectedFailedRevision int32, expectedFallbackReason, expectedFallbackMessage, nodeName string) {
```

Comment:
I don't think we have to check the message. It makes maintenance hard. We have unit tests for that.

If you really want to specific, just check for a keyword in the string (strings.Contains).

Reply:
what's sure as well, there might be cases that would break etcd, I can remove.